catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/DecimalRangeValidator.java

```diff
@@ -0,0 +1,88 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.validation.impl.validator;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Set;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.validation.AttributeValidator;
+import ddf.catalog.validation.impl.report.AttributeValidationReportImpl;
+import ddf.catalog.validation.impl.violation.ValidationViolationImpl;
+import ddf.catalog.validation.report.AttributeValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+import ddf.catalog.validation.violation.ValidationViolation.Severity;
+
+public class DecimalRangeValidator implements AttributeValidator {
+    private final double min;
+
+    private final double max;
+
+    private final double epsilon;
+
+    public DecimalRangeValidator(final double min, final double max) {
+        this(min, max, 1e-6);
+    }
+
+    public DecimalRangeValidator(final double min, final double max, final double epsilon) {
+        this.min = min;
+        this.max = max;
+        this.epsilon = epsilon;
+    }
+
+    @Override
+    public AttributeValidationReport validate(Attribute attribute) {
+        final AttributeValidationReport report = new AttributeValidationReportImpl();
+
+        if (attribute != null) {
+            final String name = attribute.getName();
+            for (final Serializable value : attribute.getValues()) {
+                final double doubleValue;
+                if (value instanceof Short) {
+                    doubleValue = ((Short) value).doubleValue();
+                } else if (value instanceof Integer) {
+                    doubleValue = ((Integer) value).doubleValue();
+                } else if (value instanceof Long) {
+                    doubleValue = ((Long) value).doubleValue();
+                } else if (value instanceof Float) {
+                    doubleValue = ((Float) value).doubleValue();
+                } else if (value instanceof Double) {
+                    doubleValue = (double) value;
+                } else {
+                    continue;
+                }
+
+                if (!checkRange(doubleValue)) {
+                    final String violationMessage = String.format("%s must be between %f and %f",
+                            name,
+                            min,
+                            max);
+                    final Set<ValidationViolation> violations =
+                            report.getAttributeValidationViolations();
+                    violations.add(new ValidationViolationImpl(Collections.singleton(name),
+                            violationMessage,
+                            Severity.ERROR));
+                    break;
+                }
+            }
+        }
+
+        return report;
```

Comment:
I don't know. We've talked about it a bit, but I'm not sure if it makes sense to suggest values for something like a decimal range because we have no idea what "good" suggested values would be. I implemented suggested values for the enumeration validator because I feel like that's the only type of validation for which we can confidently suggest values.